editor/filesystem_dock.cpp

```diff
@@ -1562,7 +1562,12 @@ void FileSystemDock::_folder_removed(String p_folder) {
 }
 
 void FileSystemDock::_rename_operation_confirm() {
-	String new_name = rename_dialog_text->get_text().strip_edges();
+	if (!tree->is_anything_selected()) {
+		return;
+	}
```

Comment:
I think this condition is not necessary (`is_anything_selected()` is a complex method). You could just do `get_selected()` and check if it's not null.

(although I haven't tested if that would work as expected, so maybe I'm wrong)

Reply:
That might be a different function. The current one is a null comparison.
`bool Tree::is_anything_selected() {
     return (selected_item != nullptr);
}`

Reply:
You are right, I looked at wrong function 🤦‍♂️